x-template.js

```diff
@@ -269,6 +270,1104 @@ class Forgiving {
+  // “<” character.
+  static #unboundContent = /[^<]+/y;
+
+  // Comments start with “<!--” and go until the next occurrence of “-->”. Note
+  // that we use non-greedy matching and we allow the “.” to match newlines.
+  static #unboundComment = /<!--.*?-->/ys;
+
+  // Tag names can contain the lowercase characters [a-z], [0-9], and a hyphen.
+  // But, the first character must be [a-z] and the last character cannot be a
+  // hyphen. The first character of the tag must directly follow the opening
+  // angle bracket. The closing angle bracket of the open tag cannot be preceded
+  // by a space or newline. The closing tag must not contain spaces or newlines.
+  // - ok: <h6>, <my-element-1>
+  // - not ok: <-div>, <1-my-element>
+  static #openTagStart = /<(?![0-9-])[a-z0-9-]+(?<!-)(?=[\s\n>])/y;
+  static #closeTag =   /<\/(?![0-9-])[a-z0-9-]+(?<!-)>/y;
+  static #openTagSpace =   /[\s\n]+/y;
+  static #openTagEnd = /(?<![\s\n])>/y;
+
+  // Attribute names can contain the characters [a-z], [A-Z], [0-9], and a
+  // hyphen. But, they cannot begin with numbers, capital letters, or hyphens —
+  // and they cannot end in a hyphen.
+  // - ok: x1, foo-bar, viewBox
+  // - not ok: 1a, Hi, -id, title-
+  static #unboundBoolean =   /(?![A-Z0-9-])[a-zA-Z0-9-]+(?<!-)(?=[\s\n>])/y;
+  static #unboundAttribute = /(?![A-Z0-9-])[a-zA-Z0-9-]+(?<!-)="[^"]*"(?=[\s\n>])/y;
+  static #boundBoolean =   /\?(?![A-Z0-9-])[a-zA-Z0-9-]+(?<!-)="$/y;
+  static #boundDefined = /\?\?(?![A-Z0-9-])[a-zA-Z0-9-]+(?<!-)="$/y;
+  static #boundAttribute =   /(?![A-Z0-9-])[a-zA-Z0-9-]+(?<!-)="$/y;
+
+  // Property names can contain the characters [a-z], [A-Z], [0-9], and an
+  // underscore. But, they cannot begin with numbers, capital letters, or
+  // underscores — and they cannot end with an underscore.
+  // - ok: id, className, defaultValue
+  // - not ok: snake_case, YELLING, 1a, _private
+  static #boundProperty = /\.(?![A-Z0-9_])[a-zA-Z0-9_]+(?<!_)="$/y;
+
+  // We require that values bound to attributes and properties be enclosed
+  // in double-quotes (see above patterns). Because interpolations delimit our
+  // “strings”, we need to check that the _next_ string begins with a
+  // double-quote. Note that it must precede a space or the closing angle
+  // bracket of the opening tag.
+  static #danglingQuote = /"(?=[\s\n>])/y;
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Special Tag Patterns //////////////////////////////////////////////////////
+  //////////////////////////////////////////////////////////////////////////////
+
+  // The “textarea” and “style” tags are special in that they contain so-called
+  // “replaceable” and “non-replaceable” character data. In either case, we
+  // must begin by treating all characters between the opening and closing tags
+  // as their content. Note that we allow the “.” to match across newlines.
+  static #throughStyle =    /.*?<\/style>/ys;
```

Comment:
nit: note that `style` will be deprecated

Reply:
This is kinda obviated IMO. Not going to worry about it for now.